swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/work/WorkQueueManager.kt

```diff
@@ -48,37 +47,28 @@ class WorkQueueManager(
   private val clock: Clock,
   private val registry: Registry,
   private val cacheStatus: CacheStatus
-) : DiscoveryActivated {
+) : DiscoveryActivated() {
   private val log: Logger = LoggerFactory.getLogger(javaClass)
   private val queueId = registry.createId("swabbie.redis.queue")
 
-  private val up: AtomicBoolean = AtomicBoolean()
-
   init {
     queue.track()
   }
 
-  override val onDiscoveryUpCallback: (event: RemoteStatusChangedEvent) -> Unit
-    get() = {
-      up.set(true)
-      if (isEnabled()) {
-        ensureLoadedCaches()
-        queue.refillOnEmpty()
-      }
-    }
-
-  override val onDiscoveryDownCallback: (event: RemoteStatusChangedEvent) -> Unit
-    get() = {
-      up.set(false)
+  override fun onDiscoveryUpCallback(event: RemoteStatusChangedEvent) {
+    if (isEnabled()) {
```

Comment:
as the code is now, this `override` makes it so `DiscoveryActivated` never does the up/down business. see my earlier comment in that class.